hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/updatetokencustomfees/UpdateTokenCustomFeesDecoder.java

```diff
@@ -162,52 +157,60 @@ private Stream<CustomFee> decodeFractionalFees(
+                        .fallbackFee(getFallbackFee(attempt, fee))
                         .build())
-                .feeCollectorAccountId(addressIdConverter.convert(fee.get(ROYALTY_FEE_FEE_COLLECTOR)))
+                .feeCollectorAccountId(attempt.addressIdConverter().convert(fee.get(ROYALTY_FEE_FEE_COLLECTOR)))
                 .build());
     }
 
     // In the solidity structure for Fixed Fees we have the property `bool useCurrentTokenForPayment` that is not
     // present in the protobuf version.
     // In order to determine the denominating token we need to check if the property is present.
-    private TokenID determineDenominatingToken(@NonNull final Tuple fee, @NonNull final TokenID tokenId) {
+    private TokenID determineDenominatingToken(
+            @NonNull final HtsCallAttempt attempt, @NonNull final Tuple fee, @NonNull final TokenID tokenId) {
         final boolean useCurrentToken = fee.get(FIXED_FEE_USE_CURRENT_TOKEN_FOR_PAYMENT);
         return useCurrentToken
                 ? tokenId
-                : getDenominationTokenIdOrNull(fee.get(FIXED_FEE_TOKEN_ID), fee.get(FIXED_FEE_USE_HBARS_FOR_PAYMENT));
+                : getDenominationTokenIdOrNull(
+                        attempt, fee.get(FIXED_FEE_TOKEN_ID), fee.get(FIXED_FEE_USE_HBARS_FOR_PAYMENT));
     }
 
     // In the solidity structure for FixedFees we have the property `bool useHbarsForPayment` that is not present
     // in the protobuf version where we set the denominating token to null if the fee is in hbars and/or we provide zero
     // address.
     private @Nullable TokenID getDenominationTokenIdOrNull(
-            @NonNull final Address address, final boolean useHbarsForPayment) {
-        final var tokenId = ConversionUtils.asTokenId(address);
-        return useHbarsForPayment || tokenId.equals(TokenID.DEFAULT) ? null : tokenId;
+            @NonNull final HtsCallAttempt attempt, @NonNull final Address address, final boolean useHbarsForPayment) {
+        final var tokenId = ConversionUtils.asTokenId(attempt.nativeOperations().entityIdFactory(), address);
+        return useHbarsForPayment
+                        || tokenId.equals(
+                                attempt.nativeOperations().entityIdFactory().newTokenId(0))
+                ? null
+                : tokenId;
     }
 
-    private @Nullable FixedFee getFallbackFee(@NonNull Tuple fee) {
+    private @Nullable FixedFee getFallbackFee(@NonNull final HtsCallAttempt attempt, @NonNull Tuple fee) {
         final Address tokenAddress = fee.get(ROYALTY_FALLBACK_FEE_TOKEN_ID);
         final long amount = fee.get(ROYALTY_FALLBACK_FEE_AMOUNT);
-        return ConversionUtils.asTokenId(tokenAddress).equals(TokenID.DEFAULT) && amount == 0
+        return ConversionUtils.asTokenId(attempt.nativeOperations().entityIdFactory(), tokenAddress)
+                                .equals(TokenID.DEFAULT)
+                        && amount == 0
```

Comment:
This change itself makes sense, but is the method's logic actually correct? It could end up comparing token `5.6.0` to token `0.0.0`, which would evaluate to false, but seems like maybe the intent is actually just to compare the token numbers? 

For more context, I'm not quite clear on `TokenID.DEFAULT`'s semantic meaning when the shard/realm aren't `0.0`. Is `TokenID.DEFAULT` the _only_ sentinel value, i.e. for shard=realm=0 (`0.0.0`), or is `1.2.0` also considered a sentinel value for a node using shard=1,realm=2?

Reply:
This is really the way that the headlong library decodes the tuple of the address that are not passed in.  In this case, asTokenID would always return the `0.0.0` version of the sentinel address.